RecoTracker/LSTCore/standalone/code/core/write_lst_ntuple.cc

```diff
@@ -579,6 +621,146 @@ void setPixelTripletOutputBranches(lst::Event<Acc3D>* event) {
   ana.tx->setBranch<std::vector<int>>("pT3_isDuplicate", pT3_isDuplicate);
 }
 
+//________________________________________________________________________________________________________________________________
+void fillT5DNNBranches(lst::Event<Acc3D>* event, unsigned int T3) {
+  // Get relevant information
+  lst::Hits const* Hits = event->getHits()->data();
+  lst::Modules const* Modules = event->getModules()->data();
+
+  // Hits
+  std::vector<unsigned int> hits = getHitsFromT3(event, T3);
+  unsigned int hit0 = hits[0];
+  unsigned int hit2 = hits[2];
+  unsigned int hit4 = hits[4];
+
+  // Hit locations
+  const float hit0_x = Hits->xs[hit0];
+  const float hit0_y = Hits->ys[hit0];
+  const float hit0_z = Hits->zs[hit0];
+  const float hit0_r = sqrt(hit0_x * hit0_x + hit0_y * hit0_y);
+  lst_math::Hit hitA(hit0_x, hit0_y, hit0_z);
+  const float hit2_x = Hits->xs[hit2];
+  const float hit2_y = Hits->ys[hit2];
+  const float hit2_z = Hits->zs[hit2];
+  const float hit2_r = sqrt(hit2_x * hit2_x + hit2_y * hit2_y);
+  lst_math::Hit hitB(hit2_x, hit2_y, hit2_z);
+  const float hit4_x = Hits->xs[hit4];
+  const float hit4_y = Hits->ys[hit4];
+  const float hit4_z = Hits->zs[hit4];
+  const float hit4_r = sqrt(hit4_x * hit4_x + hit4_y * hit4_y);
+  lst_math::Hit hitC(hit4_x, hit4_y, hit4_z);
+
+  ana.tx->pushbackToBranch<float>("t3_0_r", hit0_r);
+  ana.tx->pushbackToBranch<float>("t3_0_x", hit0_x);
+  ana.tx->pushbackToBranch<float>("t3_0_y", hit0_y);
+  ana.tx->pushbackToBranch<float>("t3_0_z", hit0_z);
+  ana.tx->pushbackToBranch<float>("t3_0_eta", hitA.eta());
+  ana.tx->pushbackToBranch<float>("t3_0_phi", hitA.phi());
+  ana.tx->pushbackToBranch<float>("t3_2_r", hit2_r);
```

Comment:
are fixed size arrays accepted to make the code less repetitive and just use 3-step loops to fill all values (to write/read to `t3_z[3]` instead of a rolled out "array")?
I can imagine trying to add non-anchor hits, which can add x2 more effort